### PR TITLE
[@chec/commerce.js] Add missing commerce.request() method

### PR DESCRIPTION
**Pull request by PaitoAnderson**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/chec/commerce.js/blob/master/src/commerce.js#L65
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Review by robbieaverill — APPROVED** [review]
This method wasn't intended to be consumed directly via the Commerce.js API, however I have no problem with adding it to the definitions

**Comment by PaitoAnderson** [comment]
@robbieaverill Oh okay, thanks for clarifying that! 👍

File: types/chec__commerce.js/types/product.d.ts [hunk 1]
```diff
@@ -52,8 +64,9 @@ export interface Product {
     checkout: string;
     display: string;
   };
-  variants: Variant[];
+  extra_fields: any[];
+  variant_groups: Variant[];
```
Comment (robbieaverill) [inline types/chec__commerce.js/types/product.d.ts:68]: These are a list of variant groups now. We probably need to update those types to suit the latest API version. A quick summary would be:

Product's `variant_groups: VariantGroup[]`:

* id: string
* name: string
* meta: object|array ?
* created: int?
* updated: int?
* options: VariantOption[]

A Variant is a type, but it's not connected directly in product responses any more. Instead, there's a separate API to fetch variants for a product: https://commercejs.com/docs/sdk/products#get-variants. This API returns a list of Variants, and a Variant object looks like this:

* id: stirng
* sku: string
* description: string
* inventory: int?
* price: Price?
* is_valid: boolean
* invalid_reason_code: string
* meta: object|array ?
* created: int?
* updated: int?
* options: VariantOption[]
* assets: Asset[]

The `options` attribute in a variant object lists the variant options that make up this variant, for example if the product was "New season T-shirt", the variant might be "Small + Red" and the variant options would be "Small" and "Red".

Would you be keen to add in these new types?

**Review by robbieaverill — CHANGES_REQUESTED** [review]
Thanks for much for taking the time to update these types. I've left a comment around variants.

File: types/chec__commerce.js/features/products.d.ts [hunk 2]
```diff
@@ -1,17 +1,23 @@
...
+
 export class Products {
   constructor(commerce: Commerce);
 
   list(params?: any): Promise<ProductCollection>;
   retrieve(id: string, data?: object): Promise<Product>;
-  getVariants(id: string, data?: object): Promise<any>;
-  getVariant(id: string, variantId: string): Promise<any>;
+  getVariants(id: string, data?: object): Promise<VariantOptionCollection>;
+  getVariant(id: string, variantId: string): Promise<VariantOption>;
```
Comment (robbieaverill) [inline types/chec__commerce.js/features/products.d.ts:22]: These return a list of Variant models and a single Variant, not a variant option

**Review by robbieaverill — CHANGES_REQUESTED** [review]
I'm happy to help with this if you'd like

**Comment by PaitoAnderson** [comment]
Thanks! Sorry, I was struggling with naming.

**Review by robbieaverill — APPROVED** [review]
LGTM, thanks for doing those updates 🙂

**Comment by PaitoAnderson** [comment]
Ready to merge

**Pull request merged**